### DISCUSSION TITLE
How to load proxy configuration from both the configuration as from code?

### DISCUSSION THREAD

**Discussion by PieterBoeren** [dotnet/yarp#516] [Q&A] [answered]
I'm trying to combine loading configuration from both the configuration file (LoadFromConfig(..)) as from code (using the supplied "LoadFromMemory(..)" sample, but this is not working at all (routes from code are not loaded). Is this scenario supported (I want to create some "convention" based routing instead of doing a lot of configuration in the settings file)?

` services`
     `.AddReverseProxy()`
     `.LoadFromMemory(routes, clusters)`
     `.LoadFromConfig(Configuration.GetSection("ReverseProxy"));`

**Comment by Tratcher** [comment, marked as answer]
Only one source is currently supported.

For convention based config we do have `IProxyConfigFilter` that you can implement and have run on each provided cluster or route, either from code or config.
https://github.com/microsoft/reverse-proxy/blob/3832fd34bb9bdfb25fb299767420939096a590a3/samples/ReverseProxy.Config.Sample/CustomConfigFilter.cs#L13-L15

Will that work for you?

**Comment by PieterBoeren** [comment]
Probably not entirely ;) This solution will still require me to define all routes and clusters (in config) and then applying some "default" attributes on them, based on some condition.

I'm more looking for a solution where I can say (in the config file): I have app1, app2 and app3 (which are frontend applications) and then create the correct routes and clusters in code (based on some custom code off course) and combine them with the existing (API endpoints etc) routes/clusters from the (same) config file.

Is multiple configuration sources somewhere on the roadmap?

**Reply by Tratcher** [reply]
We've discussed the possibility but it's not on the roadmap yet. You can request it by opening up a new [issue](https://github.com/microsoft/reverse-proxy/issues) and detailing your requirements.

**Reply by samsp-msft** [reply]
If you don't do `.LoadFromConfig(Configuration.GetSection("ReverseProxy"));` then presumably you could use the same config section and definition, and read it from your code, augment the data with the defaults you desire and push it as a code based config?

**Reply by PieterBoeren** [reply]
I'll stick to config based for now, if it becomes too complex, I'll give that a try.
